include/ignition/sensors/ForceTorqueSensor.hh

```diff
@@ -0,0 +1,110 @@
+ * limitations under the License.
+ *
+*/
+
+#ifndef IGNITION_SENSORS_FORCETORQUESENSOR_HH_
+#define IGNITION_SENSORS_FORCETORQUESENSOR_HH_
+
+#include <memory>
+
+#include <sdf/sdf.hh>
+
+#include <ignition/common/SuppressWarning.hh>
+#include <ignition/common/Time.hh>
+#include <ignition/math/Pose3.hh>
+
+#include <ignition/sensors/config.hh>
+
+#include <ignition/sensors/force_torque/Export.hh>
+
+#include "ignition/sensors/Sensor.hh"
+
+namespace ignition
+{
+  namespace sensors
+  {
+    // Inline bracket to help doxygen filtering.
+    inline namespace IGNITION_SENSORS_VERSION_NAMESPACE {
+    //
+    /// \brief forward declarations
+    class ForceTorqueSensorPrivate;
+
+    /// \brief Force Torque Sensor Class
+    ///
+    /// A force-torque Sensor that reports force and torque applied on a joint.
+
+    class IGNITION_SENSORS_FORCE_TORQUE_VISIBLE ForceTorqueSensor : public Sensor
+    {
+      /// \brief constructor
+      public: ForceTorqueSensor();
+
+      /// \brief destructor
+      public: virtual ~ForceTorqueSensor();
+
+      /// \brief Load the sensor based on data from an sdf::Sensor object.
+      /// \param[in] _sdf SDF Sensor parameters.
+      /// \return true if loading was successful
+      public: virtual bool Load(const sdf::Sensor &_sdf) override;
+
+      /// \brief Load the sensor with SDF parameters.
+      /// \param[in] _sdf SDF Sensor parameters.
+      /// \return true if loading was successful
+      public: virtual bool Load(sdf::ElementPtr _sdf) override;
+
+      /// \brief Initialize values in the sensor
+      /// \return True on success
+      public: virtual bool Init() override;
+
+      /// \brief Update the sensor and generate data
+      /// \param[in] _now The current time
+      /// \return true if the update was successfull
+      public: virtual bool IGN_DEPRECATED(4) Update(
+        const ignition::common::Time &_now) override;
+
+      /// \brief Update the sensor and generate data
+      /// \param[in] _now The current time
+      /// \return true if the update was successfull
+      public: virtual bool Update(
+        const std::chrono::steady_clock::duration &_now) override;
+
+      /// \brief Get the current joint force.
+      /// \return The latested measured force.
+      public: math::Vector3d Force() const;
+
+      /// \brief Set the force vector.
+      /// \param[in] _force force vector in newton.
+      public: void SetForce(const math::Vector3d &_force);
+
+      /// \brief Get the current joint torque.
+      /// \return The latested measured torque.
+      public: math::Vector3d Torque() const;
+
+      /// \brief Set the torque vector.
+      /// \param[in] _torque torque vector in newton.
+      public: void SetTorque(const math::Vector3d &_torque);
+    
+
+      IGN_COMMON_WARN_IGNORE__DLL_INTERFACE_MISSING
+      /// \brief Data pointer for private data
+      /// \internal
+      private: std::unique_ptr<ForceTorqueSensorPrivate> dataPtr;
```

Comment:
you can use here `IGN_UTILS_IMPL_PTR` from `#include <ignition/utils/ImplPtr.hh>`

Reply:
`ign-sensors` doesn't directly depend on `ign-utils` yet, so maybe we can revisit this later?

Reply:
I agree, we can defer for now.

I propose we make everything depend on utils at the very beginning of the next cycle to avoid this next time...

Reply:
https://github.com/ignitionrobotics/ign-sensors/issues/182